sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -208,21 +208,25 @@ public virtual void DeleteEntity(string rowKey, ETag ifMatch = default)
 
         internal async Task<Response<TableBatchResponse>> SubmitBatchAsyncInternal(bool async, CancellationToken cancellationToken = default)
         {
-            if (_submitted)
-            {
-                throw new InvalidOperationException(TableConstants.ExceptionMessages.BatchCanOnlyBeSubmittedOnce);
-            }
-            else
-            {
-                _submitted = true;
-            }
-
-            _submittedMessageList = BuildOrderedBatchRequests();
-
             using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableTransactionalBatch)}.{nameof(SubmitBatch)}");
             scope.Start();
             try
             {
+                if (!_requestMessages.Any())
```

Comment:
Not sure if it matters to you here, but using LINQ weakens the concurrent guarantee.   You may want to consider `TryPeek` here with a discard.